### PR TITLE
Generic ERC4626 <> ERC20 Token Wrapper

### PR DESCRIPTION
**Pull request by CheyenneAtapour**

### CODE REVIEW

File: test/Generic4626Wrapper.t.sol [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address constant POOL_CONFIGURATOR =
+    0x64b761D848206f447Fe2dd461b0c635Ec39EbB27;
+  address constant ADMIN = 0x5300A1a15135EA4dc7aD5a167152C01EFc9b192A;
+  address constant AAVE_ORACLE = 0x54586bE62E3c3580375aE3723C145253060Ca0C2;
+  MockERC20 USDSToken;
+  MockERC4626 SUSDSToken;
+  address USDS;
+  address SUSDS;
+
```
Comment (miguelmtzinf) [inline test/Generic4626Wrapper.t.sol]: better to rename

File: src/Generic4626Wrapper.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  /// @inheritdoc BaseTokenWrapper
+  function borrowTokenWithPermit(
+    uint256 amount,
+    uint16 referralCode,
+    PermitSignature calldata signature
+  ) external override {
+    if (signature.deadline != 0) {
+      address debtToken = IPool(0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2)
```
Comment (miguelmtzinf) [inline src/Generic4626Wrapper.sol]: hardcoded

File: src/Generic4626Wrapper.sol [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        .getReserveData(TOKEN_OUT)
+        .variableDebtTokenAddress;
+
+      ICreditDelegationToken(debtToken).delegationWithSig(
```
Comment (miguelmtzinf) [inline src/Generic4626Wrapper.sol]: wrap in a try-catch (same issue as https://github.com/aave/token-wrappers/pull/12)

File: src/Generic4626Wrapper.sol [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    IERC20(tokenIn).approve(tokenOut, type(uint256).max);
+  }
+
+  // @inheritdoc BaseTokenWrapper
+  function borrowToken(uint256 amount, uint16 referralCode) external override {
+    _borrowToken(amount, msg.sender, referralCode);
+  }
+
+  /// @inheritdoc BaseTokenWrapper
+  function borrowTokenWithPermit(
```
Comment (miguelmtzinf) [inline src/Generic4626Wrapper.sol]: I d move this to the base contract

File: test/Generic4626Wrapper.t.sol (hunk 1 above)
Comment (CheyenneAtapour) [reply test/Generic4626Wrapper.t.sol]: Done [f8b02c0](https://github.com/aave/token-wrappers/commit/f8b02c016564751f9c40f3e236bd3bee738b94d5)

File: src/Generic4626Wrapper.sol (hunk 2 above)
Comment (CheyenneAtapour) [reply src/Generic4626Wrapper.sol]: Fixed [f8b02c0](https://github.com/aave/token-wrappers/commit/f8b02c016564751f9c40f3e236bd3bee738b94d5)

File: src/Generic4626Wrapper.sol (hunk 3 above)
Comment (CheyenneAtapour) [reply src/Generic4626Wrapper.sol]: Done [f8b02c0](https://github.com/aave/token-wrappers/commit/f8b02c016564751f9c40f3e236bd3bee738b94d5)

File: src/Generic4626Wrapper.sol (hunk 4 above)
Comment (CheyenneAtapour) [reply src/Generic4626Wrapper.sol]: Done [f8b02c0](https://github.com/aave/token-wrappers/commit/f8b02c016564751f9c40f3e236bd3bee738b94d5)

File: test/SavingUsdsTokenWrapper.t.sol
Comment (miguelmtzinf) [inline test/SavingUsdsTokenWrapper.t.sol:1]: we can remove this one

File: test/Generic4626Wrapper.t.sol [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+import {MockERC20} from './mocks/MockERC20.sol';
+import {SigUtils} from './utils/SigUtils.sol';
+import {BaseTokenWrapperTest} from './BaseTokenWrapper.t.sol';
+
+contract Generic4626WrapperTest is BaseTokenWrapperTest {
+  address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address constant POOL_CONFIGURATOR =
+    0x64b761D848206f447Fe2dd461b0c635Ec39EbB27;
+  address constant ADMIN = 0x5300A1a15135EA4dc7aD5a167152C01EFc9b192A;
+  address constant AAVE_ORACLE = 0x54586bE62E3c3580375aE3723C145253060Ca0C2;
```
Comment (miguelmtzinf) [inline test/Generic4626Wrapper.t.sol]: use aave-address-book

File: test/SavingUsdsTokenWrapper.t.sol
Comment (CheyenneAtapour) [reply test/SavingUsdsTokenWrapper.t.sol:1]: Done [5337bc5](https://github.com/aave/token-wrappers/commit/5337bc5205fac77d7b9b000f5bdfeded52cf51db)

File: test/Generic4626Wrapper.t.sol (hunk 5 above)
Comment (CheyenneAtapour) [reply test/Generic4626Wrapper.t.sol]: Done [5337bc5](https://github.com/aave/token-wrappers/commit/5337bc5205fac77d7b9b000f5bdfeded52cf51db)

**Review by miguelmtzinf — APPROVED** [review]

**Pull request merged**